### PR TITLE
Google API and destination features

### PR DESCRIPTION
**Pull request by kevinlin505**

### CODE REVIEW

**Comment by kevinlin505** [comment]
@yongbingao take a look

File: src/data/trip.js [hunk 1]
```diff
@@ -35,22 +45,26 @@ export default function trip() {
...
+      const batch = db.batch();
+      const userRef = db.collection('users').doc(currentUser.uid);
+      const tripRef = db.collection('trips').doc(tripId);
+
+      batch.update(tripRef, {
+        attendees: firebase.firestore.FieldValue.arrayUnion(userRef),
+      });
+      batch.update(userRef, {
+        trips: firebase.firestore.FieldValue.arrayUnion(tripRef),
+      });
```
Comment (kevinlin505) [inline src/data/trip.js]: @yongbingao I updated your code to combine both the update on the trip and the user profile when joining a trip. This separates the logic from updating a regular user profile or trip details.

File: src/components/UserPage/CreateTrip/CreateTrip.js [hunk 2]
```diff
@@ -42,10 +63,59 @@ export const CreateTrip = ({ actions, toggleCreateTripModal }) => {
...
+          );
+          destinationDetail.place_id = data[0].place_id;
+          destinationDetail.types = data[0].types;
+          destinationDetail.formatted_address = data[0].formatted_address;
+        }
+
+        setValues({
+          ...form,
+          destinations: [...form.destinations, destinationDetail],
+        });
```
Comment (yongbingao) [inline src/components/UserPage/CreateTrip/CreateTrip.js]: how does this set state function work? where is the ...form coming from?

File: src/hooks/useDestinationsHandler.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  useEffect(() => {
+    if (isGoogleApiReady && window.google) {
+      setPlaceServices(new window.google.maps.places.PlacesService(mapRef));
+      setAutoComplete(
+        new window.google.maps.places.Autocomplete(destinationInputRef),
+      );
+    }
+  }, [isGoogleApiReady]);
+
```
Comment (yongbingao) [inline src/hooks/useDestinationsHandler.js]: Where does window.google come from?

File: src/components/UserPage/CreateTrip/CreateTrip.js (hunk 2 above)
Comment (kevinlin505) [reply src/components/UserPage/CreateTrip/CreateTrip.js]: form is this: https://github.com/kevinlin505/goplan/pull/13/files#diff-a47ac536195e19af9d72d9ca63375f2aR42

Comment (kevinlin505) [reply src/components/UserPage/CreateTrip/CreateTrip.js]: Basically, I am updating the destinations array.

File: src/hooks/useDestinationsHandler.js (hunk 3 above)
Comment (kevinlin505) [reply src/hooks/useDestinationsHandler.js]: it's a global object created from the Google script `https://maps.googleapis.com/maps/api/js?key=${Keys.FIREBASE.apiKey}&libraries=places&sessiontoken=${auth.userId}`. Not sure if they have a lib for this that I can use.

File: src/components/UserPage/CreateTrip/CreateTrip.js (hunk 2 above)
Comment (yongbingao) [reply src/components/UserPage/CreateTrip/CreateTrip.js]: got it. makes a lot more sense knowing what form and destructure do

**Pull request merged**